packages/core/src/internal/deployment-loader/types.ts

```diff
@@ -19,7 +19,7 @@ export interface DeploymentLoader {
     contractName: string,
     artifact: Artifact
   ): Promise<void>;
-  storeBuildInfo(buildInfo: BuildInfo): Promise<void>;
+  storeBuildInfo(futureId: string, buildInfo: BuildInfo): Promise<void>;
```

Comment:
To track the link between future and build info, the `storeBuildInfo` is enhanced with the future id.

Note we are calling store build info multiple times if more than one artifact has the same build info. But that is no different than before this change.

Reply:
Makes sense. We can optimize it later if needed.